### ISSUE TITLE
RA, D2k and TD CTD

### ISSUE THREAD

**Issue by Walkman100** [OpenRA/OpenRA#15904]
### Issue Summary
Just updated to `release-20180923`, and Red Alert mod crashes to desktop, and both D2k and TD crash when clicking "Manage Content". 

RA Exception:
> Exception of type `System.InvalidOperationException`: Unit `ca` does not have any sequences defined.

D2k and TD exception (they are the same):
> Exception of type `System.Collections.Generic.KeyNotFoundException`: The given key was not present in the dictionary.

#### System Information
- **Operating System:** Windows 10 Pro version 1803 build 17134.407
- **.NET / Mono Version:** I think .NET 4.7.1 but don't know how to be sure
- **OpenRA Version:** release-20180923
- **Mod:** Red Alert, Tiberian Dawn, Dune2000

#### Additional Information:
- Steps to reproduce
	1. Updated to release-20180923 (I think I was on release-20180307 previously but not certain)
	2. Launch RA
	3. CTD

- Logs
I wiped the logs folder, launched RA and let it crash, then launched D2k and clicked Manage Content, then TD and clicked Manage Content:
[Logs.zip](https://github.com/OpenRA/OpenRA/files/2672249/Logs.zip)

- OpenRA Replays
N/A

- Screenshots & Videos
I think this is N/A

**Comment by Mailaender** [comment]
Full stacktraces:

```
OpenRA engine version release-20180923
Mod Content Manager mod version release-20180923
on map cda847dca0c3256cb7d78e42fe0e4226d49b470a (Blank Shellmap by Paul Chote).
Date: 2018-12-12 14:38:41Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.Collections.Generic.KeyNotFoundException`: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at OpenRA.Graphics.CursorProvider..ctor(ModData modData)
   at OpenRA.ModData.InitializeLoaders(IReadOnlyFileSystem fileSystem)
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Mods.Common.Widgets.Logic.MainMenuLogic.<MainMenuLogic>c__AnonStorey0.<>m__16()
   at OpenRA.Primitives.ActionQueue.PerformActions(Int64 currentTime)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)
```

```
OpenRA engine version release-20180923
Red Alert mod version release-20180923
on map 3e888b44f7ca2707f42ad4d4dcf8e4a2b2e49805 (Desert Shellmap by Scott_NZ).
Date: 2018-12-12 14:36:03Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Unit `ca` does not have any sequences defined.
   at OpenRA.Graphics.SequenceProvider.GetSequence(String unitName, String sequenceName)
   at OpenRA.Graphics.Animation.GetSequence(String sequenceName)
   at OpenRA.Graphics.Animation.PlaySequence(String sequenceName)
   at OpenRA.Graphics.Animation.PlayRepeating(String sequenceName)
   at OpenRA.Mods.Common.Traits.Render.WithSpriteBody..ctor(ActorInitializer init, WithSpriteBodyInfo info, Func`1 baseFacing)
   at OpenRA.Mods.Common.Traits.Render.WithFacingSpriteBodyInfo.Create(ActorInitializer init)
   at OpenRA.Actor..ctor(World world, String name, TypeDictionary initDict)
   at OpenRA.World.CreateActor(Boolean addToWorld, String name, TypeDictionary initDict)
   at OpenRA.Mods.Common.Traits.SpawnMapActors.WorldLoaded(World world, WorldRenderer wr)
   at OpenRA.World.LoadComplete(WorldRenderer wr)
   at OpenRA.Game.StartGame(String mapUID, WorldType type)
   at OpenRA.Game.LoadShellMap()
   at OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.StartGame(Arguments args)
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Game.Initialize(Arguments args)
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)
```

**Comment by Mailaender** [comment]
Try deleting your content folder and try again.

**Comment by Walkman100** [comment]
It didn't work, I followed the same procedure as before: [Logs.zip](https://github.com/OpenRA/OpenRA/files/2691222/Logs.zip)
It does show the content manager for a split second before crashing though.
Full stacktrace (all three logfiles are exactly the same except for time):
```
OpenRA engine version release-20180923
Mod Content Manager mod version release-20180923
Date: 2018-12-18 15:48:27Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.Collections.Generic.KeyNotFoundException`: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at OpenRA.Graphics.CursorProvider..ctor(ModData modData)
   at OpenRA.ModData.InitializeLoaders(IReadOnlyFileSystem fileSystem)
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.BeforeLoad()
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Game.Initialize(Arguments args)
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)
```
I also tried renaming the settings.yaml file but it didn't help, also tried renaming the whole OpenRA folder in Documents - it created a new one with exactly the same exception, and a ModMetadata folder with a yaml of the mod I launched.

My guess is the installer didn't do something properly - should I try uninstalling and reinstalling, or upload the game files somewhere so someone can take a look and see what's wrong? idk if it would be useful to try and prevent something like this happening again, or if I'll be the only one to have this issue.

**Comment by Mailaender** [comment]
If you are using Windows Defender, you could try adding exceptions for ransomware protection.

**Comment by pchote** [comment]
The RA stacktrace suggests that your RA assets are corrupted (complete enough to pass the ingame tests, but missing at least the cruiser artwork).   The TD/D2k/ModContent stacktrace suggests that your OpenRA install itself is missing files.

Have you made any manual changes to the installation directory? Running invasive security software like Norton? If "no" to both of those you should try doing a full uninstall, delete the OpenRA directory in your documents folder, then reinstall from scratch.

**Comment by Walkman100** [comment]
@Mailaender: Under Ransomware protection I have Controlled folder access off, I tried temporarily disabling Real-time protection to no effect.

@pchote: I haven't made any manual changes to my install dir, but it is an upgrade of some older version - I think I uninstalled and reinstalled when the change from one EXE to multiple was made. I only have Windows Defender for real-time protection.

I have uninstalled and reinstalled, kept my old content folder - and it works fine (aaand there's a new version, which works fine too).

So it was a once-off installer bug?
